gribscan/gribscan.py

```diff
@@ -223,6 +224,15 @@ def get_time_offset(gribmessage, lean_towards="end"):
     """Calculate time offset based on GRIB definition.
 
     See: https://codes.ecmwf.int/grib/format/grib1/ctable/5/
+
+    For GRIB2 output, lean_towards now supports "start", "end" and "mid"
```

Comment:
This addition is very welcome!
However, I'd ask to implement this also for GRIB1 messages. E.g. there could be a function like
```python
def lean_timestamp(offset_start, offset_end, unit, lean_towards):
```
or similar, which would implement the logic once and would be called whenever needed.